src/arrayMethodSort.js

```diff
@@ -5,7 +5,40 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let permanent;
+    let changes = 0;
+
+    if (arguments.length === 0) {
+      do {
+        changes = 0;
+
+        for (let i = 0; i < this.length - 1; i++) {
+          if (this[i].toString() > this[i + 1].toString()) {
+            permanent = this[i];
+            this[i] = this[i + 1];
+            this[i + 1] = permanent;
+            changes++;
+          }
+        }
+      } while (changes > 0);
+    }
+
+    if (arguments.length > 0) {
+      do {
+        changes = 0;
+
+        for (let i = 0; i < this.length - 1; i++) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            permanent = this[i];
+            this[i] = this[i + 1];
+            this[i + 1] = permanent;
+            changes++;
+          }
+        }
+      } while (changes > 0);
```

Comment:
```
         if (this[i].toString() > this[i + 1].toString()) {
            permanent = this[i];
            this[i] = this[i + 1];
            this[i + 1] = permanent;
            changes++;
          }
        }
      } while (changes > 0);
    }
---------------------------
          if (compareFunction(this[i], this[i + 1]) > 0) {
            permanent = this[i];
            this[i] = this[i + 1];
            this[i + 1] = permanent;
            changes++;
          }
        }
      } while (changes > 0);
```
this code differs only in this:
```
if (this[i].toString() > this[i + 1].toString()) {

if (compareFunction(this[i], this[i + 1]) > 0) {
```

if you move `(this[i].toString() > this[i + 1].toString())` logic to you own function and put it to 
```
[].__proto__.sort2 = function(compareFunction = youOwn function) {
```
then you could get off dublicate:
```
 if (this[i].toString() > this[i + 1].toString()) {
            permanent = this[i];
            this[i] = this[i + 1];
            this[i + 1] = permanent;
            changes++;
          }
        }
      } while (changes > 0);
```
all actions will do in this one block:
```
if (compareFunction(this[i], this[i + 1]) > 0) {
            permanent = this[i];
            this[i] = this[i + 1];
            this[i + 1] = permanent;
            changes++;
          }
```